b2sdk/transfer/emerge/executor.py

```diff
@@ -326,61 +326,197 @@ def _find_unfinished_file_by_plan_id(
         legal_hold: Optional[LegalHold] = None,
         custom_upload_timestamp: Optional[int] = None,
         cache_control: Optional[str] = None,
+        log_rejections: Optional[bool] = False,
+        check_file_info_without_large_file_sha1: Optional[bool] = False,
+        eager_mode: Optional[bool] = False,
     ):
+        """
+        Search for a matching unfinished large file in the specified bucket.
+
+        This function is responsible for identifying a suitable unfinished file for resumption of upload,
+        based on the supplied parameters. It centralizes the shared logic between `_find_unfinished_file_by_plan_id`
+        and `_match_unfinished_file_if_possible`.
+
+        In case a matching file is found but has inconsistencies (for example, mismatching file info or encryption settings),
+        the 'log_rejections' parameter dictates if these mismatches should be logged.
+
+        :param bucket_id: The identifier of the bucket where the unfinished file resides.
+        :param file_name: The name of the file to be matched.
+        :param file_info: Information about the file to be uploaded.
+        :param emerge_parts_dict: A dictionary containing the parts of the file to be emerged.
+        :param encryption: The encryption settings for the file.
+        :param file_retention: The retention settings for the file, if any.
+        :param legal_hold: The legal hold status of the file, if any.
+        :param custom_upload_timestamp: The custom timestamp for the upload, if any.
+        :param cache_control: The cache control settings for the file, if any.
+        :param log_rejections: A flag indicating whether rejections should be logged.
+        :param check_file_info_without_large_file_sha1: A flag indicating whether the file information should be checked without the `large_file_sha1`.
+        :param eager_mode: A flag indicating whether the first matching file should be returned.
+        
+        :return: A tuple of the best matching unfinished file and its finished parts. If no match is found, returns `None`.
+        """
+
         file_retention = file_retention or NO_RETENTION_FILE_SETTING
-        assert 'plan_id' in file_info
         best_match_file = None
         best_match_parts = {}
         best_match_parts_len = 0
+
         for file_ in self.services.large_file.list_unfinished_large_files(
             bucket_id, prefix=file_name
         ):
             if file_.file_info != file_info:
```

Comment:
`_match_unfinished_file_if_possible` used to have `if file_.file_name != file_name: reject`, I think `_find_unfinished_file_by_plan_id` didn't have it by mistake. Unless it's included in file_info? the new approach does not have that, this might lead to difficult to detect errors, no?

Reply:
Good catch, thank you!